main.go

```diff
@@ -363,66 +351,43 @@ func FlashGDB(pkgName string, ocdOutput bool, options *compileopts.Options) erro
 				daemon.Stdout = w
 				daemon.Stderr = w
 			}
-			// Make sure the daemon doesn't receive Ctrl-C that is intended for
-			// GDB (to break the currently executing program).
-			setCommandAsDaemon(daemon)
-			// Start now, and kill it on exit.
-			err = daemon.Start()
-			if err != nil {
-				return &commandError{"failed to run", daemon.Path, err}
-			}
-			defer func() {
-				daemon.Process.Signal(os.Interrupt)
-				// Maybe we should send a .Kill() after x seconds?
-				daemon.Wait()
-			}()
 		case "qemu":
 			gdbCommands = append(gdbCommands, "target remote :1234")
 
 			// Run in an emulator.
 			args := append(config.Target.Emulator[1:], tmppath, "-s", "-S")
-			daemon := exec.Command(config.Target.Emulator[0], args...)
+			daemon = exec.Command(config.Target.Emulator[0], args...)
 			daemon.Stdout = os.Stdout
 			daemon.Stderr = os.Stderr
-
-			// Make sure the daemon doesn't receive Ctrl-C that is intended for
-			// GDB (to break the currently executing program).
-			setCommandAsDaemon(daemon)
-
-			// Start now, and kill it on exit.
-			err = daemon.Start()
-			if err != nil {
-				return &commandError{"failed to run", daemon.Path, err}
-			}
-			defer func() {
-				daemon.Process.Signal(os.Interrupt)
-				// Maybe we should send a .Kill() after x seconds?
-				daemon.Wait()
-			}()
 		case "mgba":
 			gdbCommands = append(gdbCommands, "target remote :2345")
 
 			// Run in an emulator.
 			args := append(config.Target.Emulator[1:], tmppath, "-g")
-			daemon := exec.Command(config.Target.Emulator[0], args...)
+			daemon = exec.Command(config.Target.Emulator[0], args...)
 			daemon.Stdout = os.Stdout
 			daemon.Stderr = os.Stderr
+		case "msd":
+			return errors.New("gdb is not supported for drag-and-drop programmable devices")
```

Comment:
How can you run GDB on devices such as ItsyBitsy-M0 with the debug pins attached to JLink device?

Reply:
By adding the `openocd-interface` and `openocd-target` keys to the JSON file (or I think with the `-programmer=jlink` flag). This PR doesn't change any of that, the change here is that some code got moved around.

Reply:
Ah, OK. Thanks for the clarification.